skule_vote/backend/test_ballot.py

```diff
@@ -0,0 +1,1637 @@
+                {
+                    "voter": v,
+                    "candidate": candidate1,
+                    "rank": 0,
+                    "election": officer,
+                }
+            )
+
+            all_candidate_ballots.append(
+                {
+                    "voter": v,
+                    "candidate": candidate2,
+                    "rank": 1,
+                    "election": officer,
+                }
+            )
+
+            all_candidate_ballots.append(
+                {
+                    "voter": v,
+                    "candidate": ron,
+                    "rank": 2,
+                    "election": officer,
+                }
+            )
+
+        one_candidate_ballots = []
+        for v in voters[8:11]:
+            one_candidate_ballots.append(
+                {
+                    "voter": v,
+                    "candidate": candidate1,
+                    "rank": 0,
+                    "election": officer,
+                }
+            )
+
+            all_candidate_ballots.append(
+                {
+                    "voter": v,
+                    "candidate": ron,
+                    "rank": 1,
+                    "election": officer,
+                }
+            )
+
+        for v in voters[11:12]:
+            one_candidate_ballots.append(
+                {
+                    "voter": v,
+                    "candidate": candidate2,
+                    "rank": 0,
+                    "election": officer,
+                }
+            )
+
+            all_candidate_ballots.append(
+                {
+                    "voter": v,
+                    "candidate": ron,
+                    "rank": 1,
+                    "election": officer,
+                }
+            )
+
+        ballots = all_candidate_ballots + one_candidate_ballots + ron_only_ballots
+        results = self._create_results(ballots, choices, officer)
+        self.assertEqual(results["winners"], [candidate1.name])
+        self.assertEqual(results["rounds"][0][candidate1.name], 7)
+        self.assertEqual(results["rounds"][0][candidate2.name], 1)
+        self.assertEqual(results["rounds"][0][ron.name], 4)
+        self.assertEqual(len(results["rounds"]), 1)
+        self.assertEqual(results["quota"], 7)
+        self.assertEqual(results["spoiledBallots"], NUM_SPOILED)
+        self.assertEqual(results["totalVotes"], NUM_VOTERS - NUM_SPOILED)
+
+    # CASE 2: 1 seat election with winner determined in first round + one ballot that has an error
+    def test_one_seat_winner_one_round_errored(self):
+        self._create_officer(self.election_session)
+        officer = Election.objects.filter(category="officer")[0]
+
+        choices = self._create_candidates(officer, 2)
+        ron, candidate1, candidate2 = (
+            choices[0],
+            choices[1],
+            choices[2],
+        )
+
+        NUM_VOTERS = 12
+        NUM_SPOILED = 0
+        NUM_ERRORED = 1
```

Comment:
@arminale, this constant isn't used in the test. Can you change the assertEqual to: `self.assertEqual(results["rounds"][0][candidate2.name], 2 - NUM_ERRORED)` just so we can easily see which candidate lost the vote bc of the ballot? (It might be candidate1, i'm not 100% sure cause I didn't throughly check the ballots)